unitSI/tests/test_units.py

```diff
@@ -799,137 +747,66 @@ def test_unit_discarding_functions():
     """
     Test functions that discard units.
     """
-    from unitSI.unitsafefunctions import ones_like, zeros_like
 
     values = [3 * mV, np.array([1, 2]) * mV, np.arange(12).reshape(3, 4) * mV]
     for value in values:
         assert_equal(np.sign(value), np.sign(np.asarray(value)))
-        assert_equal(zeros_like(value), np.zeros_like(np.asarray(value)))
-        assert_equal(ones_like(value), np.ones_like(np.asarray(value)))
+        assert_equal(np.zeros_like(value), np.zeros_like(np.asarray(value)))
+        assert_equal(np.ones_like(value), quantity_with_dimensions(np.ones_like(np.asarray(value)),(get_dimensions(value))))
         # Calling non-zero on a 0d array is deprecated, don't test it:
         if value.ndim > 0:
             assert_equal(np.nonzero(value), np.nonzero(np.asarray(value)))
 
-
-
-def test_unitsafe_functions():
```

Comment:
It might be worth having all these (for their numpy versions) check that they raise an error for values with dimensions, as expected